### ISSUE TITLE
Break up into shorter lessons

### ISSUE THREAD

**Issue by heratyian** [appdev-lessons/ruby-intro-running-real-programs#3]
the `gets`, `byebug`, `irb`, `require`, and `gems` sections seem a bit out of place here. I think these could be in their own lessons (maybe `require` and `gems` together). Keeping this lesson simply to setting up a codespace and running ruby files from the terminal is good enough. 

Thoughts?

Some potential new lessons:
- Handling User Input (could add a bit on ARGV)
- Debugging (using Debug gem)
- Interactive Ruby / REPL
- Importing Code

**Comment by bpurinton** [comment]
I think what we have here is a case of the balance between providing timely information in a long narrative (i.e. longer lessons jam-packed like `hello-world`, `linkinbio` or this one), and providing clear instructions for each individual part of the process.

To me, this lesson is a nice, wholistic intro to ramp up from writing Ruby code in our runnable code blocks, to beginning to wire up Ruby files in a real coding environment. This leads very well into the first Ruby web app with Sinatra in subsequent lessons (albeit after the Ruby Gym).

Rather than breaking this lesson up, I think each of the important parts of the lesson that you mention could rather use their own, dedicated "how-to-guide" type lesson, which could be a handy manual / reference with additional details. I can, however, see the removal of debugging to a different lesson, since I'm unsure how many, if any students, are actually using debugging during the Ruby Gym (and then in subsequent web apps).

This goes back to previous conversations we have had about [diataxis](https://diataxis.fr/) in our materials. 

In short, I do somewhat agree, but I think @raghubetina would have some thoughts about why he designed the lesson in such a detailed and complete narrative, and how it could be broken down or added to with supplemental lessons. We will discuss!

**Comment by heratyian** [comment]
Good points. I think this lesson makes sense when taken with the companion video. Perhaps adding more detailed follow up lessons for those specific topics makes sense.
